Bottom: df8fec293a7d9704d72949da407de43728ef5629
Top:    f6b73245e41e4c75e4ad7111075c82cb789021f3
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-04-21 02:08:35 +0100

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 37aaebb..2df8cd9 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1101,7 +1101,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
